codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Headers.java

```diff
@@ -90,6 +91,23 @@ public DefaultHttp2Headers(boolean validate) {
               validate ? HTTP2_NAME_VALIDATOR : NameValidator.NOT_NULL);
     }
 
+    /**
+     * Create a new instance.
+     * @param validate {@code true} to validate header names according to
+     * <a href="https://tools.ietf.org/html/rfc7540">rfc7540</a>. {@code false} to not validate header names.
+     * @param arraySizeHint A hint as to how large the hash data structure should be.
+     * The next positive power of two will be used. An upper bound may be enforced.
+     */
+    @SuppressWarnings("unchecked")
+    public DefaultHttp2Headers(boolean validate, int arraySizeHint) {
+        // Case sensitive compare is used because it is cheaper, and header validation can be used to catch invalid
+        // headers.
+        super(CASE_SENSITIVE_HASHER,
+              CharSequenceValueConverter.INSTANCE,
+              validate ? HTTP2_NAME_VALIDATOR : NameValidator.NOT_NULL,
+              min(arraySizeHint, 32));
```

Comment:
This is currently being used in a few spots where "estimates" are being used, and the size from http/1.x header objects are translated to http/2. I previously applied the limit in these places, but just consolidated the limit here. I'm not sure necessary to size the underlying array to the same size as number of elements. The `DefaultHeaders.index()` makes some assumptions about the size of the array...let me move this enforcement to `DefaultHeaders`